src/test/java/com/uber/cadence/internal/compatibility/thrift/HistoryMapperEventTest.java

```diff
@@ -0,0 +1,325 @@
+package com.uber.cadence.internal.compatibility.thrift;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+import com.google.protobuf.Message;
+import com.google.protobuf.Timestamp;
+import com.uber.cadence.EventType;
+import com.uber.cadence.HistoryEvent;
+import com.uber.cadence.internal.compatibility.MapperTestUtil;
+import com.uber.cadence.internal.compatibility.ProtoObjects;
+import com.uber.cadence.internal.compatibility.ThriftObjects;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.apache.thrift.TBase;
+import org.apache.thrift.TFieldIdEnum;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class HistoryMapperEventTest<
+    F extends Message, E extends Enum<E> & TFieldIdEnum, T extends TBase<T, E>> {
+  private static final Map<Class<? extends Message>, Method> ATTRIBUTE_CONVERTERS =
+      indexSingleParameterMethods(HistoryMapper.class, Message.class);
+  private static final long EVENT_ID = 1;
+  private static final int TIMESTAMP_NANOS = 100;
+  private static final Timestamp TIMESTAMP =
+      Timestamp.newBuilder().setNanos(TIMESTAMP_NANOS).build();
+  private static final long VERSION = 2;
+  private static final long TASK_ID = 3;
+
+  @Parameterized.Parameter public EventType eventType;
+
+  @Parameterized.Parameter(1)
+  public F from;
+
+  @Parameterized.Parameter(2)
+  public T to;
+
+  @Parameterized.Parameter(3)
+  public Set<String> missingFields;
+
+  @Test
+  public void testHistoryEvent() {
```

Comment:
nit: maybe test `history` constructor directly

Reply:
I'll be testing that in a followup that covers all the remaining mappers